src/timers.h

```diff
@@ -15,6 +15,21 @@ double ChildrenCPUUsage();
 // Return the CPU usage of the current thread
 double ThreadCPUUsage();
 
+#if defined(BENCHMARK_OS_QURT)
+
+// std::chrono::now() can return 0 on some Hexagon devices;
+// this reads the value of a 56-bit, 19.2MHz hardware counter
+// and converts it to seconds. Unlike std::chrono, this doesn't
+// return an absolute time, but since ChronoClockNow() is only used
+// to compute elapsed time, this shouldn't matter.
+inline double ChronoClockNow() {
```

Comment:
Could you please wrap this into a clock-like class,
and change `ChooseClockType` to pick it on `BENCHMARK_OS_QURT`?

Reply:
That seems a lot more complex than actually needed, but OK.

Reply:
Actually, I'm gonna push back on this a bit: the only reason that the `ChooseClockType` type exists is so that we can call `now()` on it in `ChronoClockNow()`, it's literally the only place in the code base that we use it. Trying to create a class that mimics the existence of `std::chrono` is pointless overhead. If I'm missing something subtle, please let me know, but this seems like a useless exercise.

Reply:
https://godbolt.org/z/T84efY5fG

Reply:
I'm not disputing what the object code will look like; I'm disputing that constructing a class would be wrong, because `std::chrono` implies a valid date/time, not just a relative time counter, and we can't mimic that with this approach; it means we have to mimic an API designed to do all sorts of arbitrary things with timestamps, rather than one that simply provides a seconds-counter in the form of a double.

Reply:
[`std::chrono::steady_clock`](https://en.cppreference.com/w/cpp/chrono/steady_clock):
> This clock is not related to wall clock time (for example, it can be time since last reboot), and is most suitable for measuring intervals.